core/src/main/java/com/novoda/noplayer/internal/exoplayer/ExoPlayerFacade.java

```diff
@@ -136,7 +135,14 @@ void loadVideo(PlayerSurfaceHolder playerSurfaceHolder,
                 bandwidthMeter
         );
         attachToSurface(playerSurfaceHolder);
-        exoPlayer.prepare(mediaSource, RESET_POSITION, DO_NOT_RESET_STATE);
+
+        boolean hasInitialPosition = options.getInitialPositionInMillis().isPresent();
+        if (hasInitialPosition) {
+            Long initialPositionInMillis = options.getInitialPositionInMillis().get();
+            exoPlayer.seekTo(initialPositionInMillis);
```

Comment:
I guess this is the fix

:+1:

Reply:
Ah yes, sorry @juankysoriano I should have commented this line 😄